src/pydata_sphinx_theme/theme/pydata_sphinx_theme/sections/announcement.html

```diff
@@ -36,15 +26,30 @@
         div.style.setProperty("min-height", "3rem");
       }, 320);
     }, 10);
-  })
-  .catch(error => {
-    console.log("[PST]: Failed to load announcement at: {{ theme_announcement }}");
-  });
+  }
+is_remote = {{ is_remote | tojson }};
+if (is_remote) {
+  document.write(`<aside class="bd-header-announcement d-print-none" aria-label="{{ banner_label }}"></aside>`);
+  fetch("{{ theme_announcement }}")
+    .then(res => {return res.text();})
+    .then(data => {
+      if (data.length === 0) {
+        console.log("[PST]: Empty announcement at: {{ theme_announcement }}");
+        return;
+      }
```

Comment:
If I'm reading this correctly, the `aside` tag will get written first, even if `data.length === 0`.  Seems like we wouldn't want that to happen?

Reply:
oh good point, let me fix it

Reply:
It looks to me that the reason the banner disappeared in #1817 is because the "init" class was getting applied to the `<aside>` tag when the announcement was local and then never removed. 

If that's the case then #1808 should fix #1817.

Reply:
By the way, PR 1808 has a test case that checks that the banner gets rendered when theme announcement is local. 

As an aside, one reason I was holding off on 1808 was that I wanted to see if someone had any thoughts about how to [write the tests I added in a more efficient way](https://github.com/pydata/pydata-sphinx-theme/pull/1808/files#r1596977639), but that could always be done in a follow-up PR.

Reply:
Oh - sorry about that, I should have checked. I'll pause here and let you finish 1808 first, thanks!

Reply:
No worries, I think it's awesome you jumped in to help fix the problem :)

Reply:
@melissawm, I edited some updates you made to the docs about the theme announcement, so if you could double check that [my rewrite](https://github.com/pydata/pydata-sphinx-theme/pull/1808/files#r1596928177) is accurate that would be awesome